modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/ColumnParams.java

```diff
@@ -204,11 +210,85 @@ public Builder length(Integer length) {
             return this;
         }
 
-        /** Builds parameters. */
+        /** Test only builder wrapper. */
+        @TestOnly
         public ColumnParams build() {
+            return build(new ColumnParamsSupplier() {
+                @Override
+                public int getMaxPrecision() {
+                    return 1 << 15;
+                }
+
+                @Override
+                public int getMaxScale() {
+                    return 10;
+                }
+            });
+        }
+
+        /** Builds parameters. */
+        public ColumnParams build(ColumnParamsSupplier helper) {
+            validate(helper);
+
             ColumnParams params0 = params;
             params = null;
             return params0;
         }
+
+        private void validate(ColumnParamsSupplier helper) {
+            CatalogParamsValidationUtils.validateIdentifier(params.name(), "Column name");
+
+            if (params.type == null) {
+                throw new CatalogValidationException(format("Column '{}' type is not specified", params.name()));
+            }
+
+            PrecisionScale precScale = params.type.precScale();
+
+            if (precScale == PrecisionScale.NO_NO) {
```

Comment:
what about `length`? as far as I can see it's still possible to create INT column with custom length, so please cover validation with exhaustive test coverage